.github/workflows/test-macos.yml

```diff
@@ -0,0 +1,23 @@
+name: test-macOS
```

Comment:
Since jenkins convers all platform test, maybe just test linux on github action is enough ?

Reply:
Running on all three platforms gives authors an early signal on all Tier 1 platforms (I could even argue that we should be testing ARM here). This is less a matter of coverage (we're not covering anything extra with Travis or GitHub Actions today), but rather a matter of early feedback on an easier to consume interface for outside contributors (this was the goal when Travis was introduced at least).

Reply:
If we add macOS, should we add windows test too ?

Reply:
I think so, yes. I just didn't add because I have no idea how we test Windows on Jenkins (the commands I used on Linux and macOS came from Jenkins config, so they are more suited for CI environments). 

If anyone knows the best way to run tests on Windows on CI, feel free to comment :)

There's always the option to add tests for Windows in a follow up.

Reply:
I'm okay with Tier 1 platforms for now. Linux is actually the slowest platform on Actions for our build/tests (* may possibly change if we start running tests on Windows). 

I'm also happy for tests to be added on Windows in a follow up PR (we're not running tests on Windows in Actions or Travis CI before this PR).